### PR TITLE
[`Blip`] Fix slow tests and doctests with correct values

### LINKED ISSUE #22625: BLIP coco base default num_attention_heads should be 12
### System Info

I was trying to replicate BLIP results on VG Relation dataset as in https://github.com/mertyg/vision-language-models-are-bows. I was using `BlipForImageTextRetrieval.from_pretrained("Salesforce/blip-itm-base-coco")` 
following this example https://huggingface.co/docs/transformers/model_doc/blip#transformers.BlipForImageTextRetrieval 

The results were much poorer, and after a lot of digging it turned out that this is because the num_attention_head in the current HuggingFace implementation is 8 instead of 12 as in https://github.com/mertyg/vision-language-models-are-bows (and as in common in BLIP).

I manually changed it via 
```
model = BlipForImageTextRetrieval.from_pretrained("Salesforce/blip-itm-base-coco")
processor = AutoProcessor.from_pretrained("Salesforce/blip-itm-base-coco")

tmp_config=model.config 
tmp_config.text_config.num_attention_heads = 12
model = BlipForImageTextRetrieval.from_pretrained(
    "Salesforce/blip-itm-base-coco", config=tmp_config
)   
```
and matched the results on VGR.  For example, if you run the doc example (with a slightly changed caption):
```
from PIL import Image
import requests
from transformers import AutoProcessor, BlipForImageTextRetrieval

model = BlipForImageTextRetrieval.from_pretrained("Salesforce/blip-itm-base-coco")
processor = AutoProcessor.from_pretrained("Salesforce/blip-itm-base-coco")

url = "http://images.cocodataset.org/val2017/000000039769.jpg"
image = Image.open(requests.get(url, stream=True).raw)
text = "two cats on a couch"

inputs = processor(images=image, text=text, return_tensors="pt")
outputs = model(**inputs)
print(torch.softmax(outputs['itm_score'],dim=-1))
```
image

**with 8 heads you get scores of `[[0.9489, 0.0511]]` for the above image i.e. the caption ` "two cats on a couch"` is not relevant, whereas with 12 heads you get scores of `[[0.1727, 0.8273]]` i.e. the caption is very relevant (which is the correct answer)**

Proposed fix:
**I suggest we switch it to 12 by default.**

### PR DESCRIPTION
**Pull request by younesbelkada**
# What does this PR do?

Fixes https://github.com/huggingface/transformers/issues/22625

In fact, the `num_attention_heads` of `BlipTextModel` should be 12 and not 8. Hence the models that are on the Hub were producing different logits / generations that the original implementation in some cases.

I made PRs on the Hub:

- https://huggingface.co/Salesforce/blip-itm-large-flickr/discussions/1
- https://huggingface.co/Salesforce/blip-itm-base-coco/discussions/3
- https://huggingface.co/Salesforce/blip-image-captioning-base/discussions/13
- https://huggingface.co/Salesforce/blip-image-captioning-large/discussions/8#642eed4cce2efe48a1aa1497
- https://huggingface.co/Salesforce/blip-vqa-base/discussions/3#642eed68ae8ae35b7a9bbb7f
- https://huggingface.co/Salesforce/blip-vqa-capfilt-large/discussions/5
- https://huggingface.co/Salesforce/blip-itm-large-flickr/discussions/3
- https://huggingface.co/Salesforce/blip-itm-large-coco/discussions/3

And tested them on the slow tests and doctests with `.from_pretrained(xxx, revision="refs/pr/xx")`, this PR fixes the slow tests and doctests with the correct values. Let's merge this PR and I'll merge the PRs that are on the Hub

cc @sgugger

### CODE REVIEW

**Review by sgugger — APPROVED** [review]
Thanks for the fixes!
